monitoring/bin/create_elasticsearch_datasource.sh

```diff
@@ -0,0 +1,203 @@
+        log_error "Monitoring needs to be deployed in this namespace in order to configure Elasticsearch with this script.";
+        exit 1
+    else
+        log_debug "Monitoring found in $MON_NS namespace.  Continuing."
+    fi
+else
+    log_info "Checking the [$tenantNS] namespace for monitoring deployment for the [$tenant] tenant ..."
+    if [[ $(kubectl get pods -n $tenantNS -l app.kubernetes.io/instance=v4m-grafana-$tenant -o custom-columns=:metadata.namespace --no-headers | uniq | wc -l) == 0 ]]; then
+        log_error "No monitoring components were found for $tenant in the $tenantNS namespace."
+        log_error "Monitoring needs to be deployed in this namespace in order to configure Elasticsearch with this script.";
+        exit 1
+    else
+        log_debug "Monitoring deployment was found for the $tenant tenant in the $tenantNS namespace.  Continuing."
+    fi
+fi
+
+# Check to see if logging namespace provided exists and components have already been deployed
+log_info "Checking for Elasticsearch pods in the $LOG_NS namespace ..."
+if [[ $(kubectl get pods -n $LOG_NS -l app=v4m-es -o custom-columns=:metadata.namespace --no-headers | uniq | wc -l) == 0 ]]; then
+  log_error "No logging components found in the $LOG_NS namespace."
+  log_error "Logging needs to be deployed in this namespace in order to configure Elasticsearch with this script.";
+  exit 1
+else
+  log_debug "Logging deployment found in $LOG_NS namespace.  Continuing."
+fi
+
+# get admin credentials
+export ES_ADMIN_USER=$(kubectl -n $LOG_NS get secret internal-user-admin -o=jsonpath="{.data.username}" |base64 --decode)
+export ES_ADMIN_PASSWD=$(kubectl -n $LOG_NS get secret internal-user-admin -o=jsonpath="{.data.password}" |base64 --decode)
+
+get_sec_api_url
+get_credentials_from_secret admin
+
+# FOR TENANT - Check to see if tenant has been deployed in logging.
+if [ "$cluster" != "true" ]; then
+    log_info "Checking that [$tenant] has been onboarded to Elasticsearch in the $LOG_NS namespace ..."
+
+    if ! kibana_tenant_exists "${tenantNS}_${tenant}"; then
+        log_error "Unable to configure Elasticsearch data source."
+        log_error "The [$tenant] tenant in the $tenantNS namespace has not been onboarded.";
+        exit 1
+    else
+        log_debug "The [$tenant] tenant has been been onboarded in the Elasticsearch.  Continuing."
+    fi 
+fi
+
+# Set user ID and password
+if [ "$cluster" == "true" ]; then
+   grfds_user="V4M_ALL_grafana_ds"
+else
+   grfds_user="${nst}_grafana_ds"
+fi
+
+if user_exists "$grfds_user"; then
+   log_verbose "Removing the existing [$grfds_user] utility account."
+   delete_user $grfds_user
+fi
+
+grfds_passwd="$(randomPassword)"
+
+if [ "$cluster" == "true" ]; then
+   ./logging/bin/user.sh CREATE -ns _all_ -t _all_ -u $grfds_user -p "$grfds_passwd" -g
+elif [ -z "$tenant" ]; then
+   ./logging/bin/user.sh CREATE -ns $tenantNS -u $grfds_user -p "$grfds_passwd" -g
```

Comment:
See notes above about whether we're going to support creating datasources limited to a namespace (but not to a specific tenant w/in that namespace).